### PR TITLE
RHIENG-9113 fixed jest config and added unit tests

### PR DESCRIPTION
**Pull request by PreetiW**
## PR Title :boom:

Migrated unit tests from Enzym to React Testing Library. 

## Why do we need this change? :thought_balloon:

Please include the __context of this change__ here.

## Documentation requires update? :memo:

- [ ] Yes
- [ ] No

## Security Checklist :lock:

Upon raising this PR please go through [RedHatInsights/secure-coding-checklist](https://github.com/RedHatInsights/secure-coding-checklist)

## :guardsman: Checklist :dart:

- [ ] Bugfix
- [ ] New Feature
- [ ] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [ ] Dependency Added

## Additional :mega:

Feel free to add any other relevant details such as __links, notes, screenshots__, here.

### CODE REVIEW

**Review by r14chandra — CHANGES_REQUESTED** [review]
Thanks for the PR, @PreetiW! I have requested some changes. Have a look.

File: src/Components/SystemDetail/RecommendationsTable.test.js [hunk 1]
```diff
@@ -1,3 +1,65 @@
-import { test } from '@jest/globals';
+// import { test } from '@jest/globals';
 
-test.todo('RecommendationsTable component tests');
+// test.todo('RecommendationsTable component tests');
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/SystemDetail/RecommendationsTable.test.js:3]: done, will push the changes in the next commit.

File: src/Components/NoEntitiesFound/NoEntitiesFound.test.js [hunk 2]
```diff
@@ -1,3 +1,23 @@
-import { test } from '@jest/globals';
+import { describe } from '@jest/globals';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import NoEntitiesFound from './NoEntitiesFound';
+
+describe('NoEntitiesFound tests', () => {
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/NoEntitiesFound/NoEntitiesFound.test.js:6]: @r14chandra I like to start the test with `describe` to specify the main work and then use `it` for different test cases and hence I used the same here. For now I am resolving the comment, feel free to open it if any changes needed 🙌🏻

File: src/Components/RecommendationRating/RecommendationRating.test.js [hunk 3]
```diff
@@ -1,3 +1,63 @@
-import { test } from '@jest/globals';
+import { describe } from '@jest/globals';
+import { render, screen } from '@testing-library/react';
+import RecommendationRating from './RecommendationRating';
+import '@testing-library/jest-dom';
 
-test.todo('RecommendationRating tests');
+describe('RecommendationRating tests', () => {
+    it('expect to render when no feedback submitted', () => {
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/RecommendationRating/RecommendationRating.test.js:7]: @r14chandra done, will push the changes in next commit. 💯

File: src/Components/RecommendationRating/RecommendationRating.test.js [hunk 4]
```diff
@@ -1,3 +1,63 @@
...
+        // render
+        const propValues = {
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/RecommendationRating/RecommendationRating.test.js:9]: @r14chandra you are correct but as the code is very small and `propValues` are different for each case I feel its okay to keep it like this. For now resolving this comment, feel free to open if needed.

File: src/Components/RecommendationRating/RecommendationRating.test.js [hunk 5]
```diff
@@ -1,3 +1,63 @@
...
+        // query
+        const thumbsUpButton = screen.getByRole('button', { name: 'thumbs-up' });
+        const thumbsDownButton = screen.getByRole('button', { name: 'thumbs-down' });
+
+        // assert
+        expect(thumbsUpButton).toBeVisible();
+        expect(thumbsDownButton).toBeVisible();
+    });
+
+    it('expect to render with positive feedback', () => {
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/RecommendationRating/RecommendationRating.test.js:26]: done, will push the changes in the next commit. 💯

File: src/Components/RecommendationRating/RecommendationRating.test.js [hunk 6]
```diff
@@ -1,3 +1,63 @@
...
+        // query & interact
+        const thumbsUpButton = screen.getByRole('button', { name: 'thumbs-up' });
+        const thumbsUpPositiveImg = screen.getByTestId('thumbs-up-positive');
+
+        // assert
+        expect(thumbsUpButton).toBeVisible();
+        expect(thumbsUpPositiveImg).toHaveClass('pf-v5-svg like');
+    });
+
+    it('expect to render with negative feedback', () => {
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/RecommendationRating/RecommendationRating.test.js:45]: done, will push the changes in the next commit. 💯

File: src/Components/Reports/Common/SystemsTablePage.test.js [hunk 7]
```diff
@@ -1,3 +1,26 @@
-import { test } from '@jest/globals';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { SystemsTablePage } from './SystemsTablePage';
 
-test.todo('Systems Table Page Component');
+describe('Systems Table Page Component', () => {
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/Reports/Common/SystemsTablePage.test.js:5]: done, will push the changes in the next commit. 💯

File: src/Components/Reports/Common/SystemsFirstPage.test.js [hunk 8]
```diff
@@ -1,3 +1,53 @@
...
+import { SystemsFirstPage } from './SystemsFirstPage';
+
+jest.mock('@redhat-cloud-services/frontend-components-pdf-generator', () => ({
+    __esModule: true,
+    Section: jest.fn(({ children, ...props }) => <div {...props}>{children}</div>),
+    Column: jest.fn(({ children, ...props }) => <div {...props}>{children}</div>),
+    Table: jest.fn(() => <div>Table inside PDF</div>)
+}));
+
+describe('Systems First Page Component', () => {
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/Reports/Common/SystemsFirstPage.test.js:12]: done, will push the changes in the next commit. 💯

File: src/Components/Reports/Common/SystemsFirstPage.test.js [hunk 9]
```diff
@@ -1,3 +1,53 @@
...
+        };
+
+        const { asFragment } = render(<SystemsFirstPage {...firstPageProps} />);
+
+        // assert
+        expect(asFragment()).toMatchSnapshot();
+
+    });
+
+    it('should matches snapshot for systems first page and renders correctly with 1 system', () => {
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/Reports/Common/SystemsFirstPage.test.js:34]: @r14chandra Done, will push the changes in the next commit. 💯

File: src/Components/Reports/Common/SystemsFirstPage.test.js [hunk 10]
```diff
@@ -1,3 +1,53 @@
-import { test } from '@jest/globals';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { SystemsFirstPage } from './SystemsFirstPage';
+
+jest.mock('@redhat-cloud-services/frontend-components-pdf-generator', () => ({
+    __esModule: true,
+    Section: jest.fn(({ children, ...props }) => <div {...props}>{children}</div>),
```
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/Reports/Common/SystemsFirstPage.test.js:7]: @r14chandra is it okay to keep the above code? For now I am resolving the comment. Feel free to open it if it is necessary to change the mock implementation. Thanks!

[the remark this answers is not in the document]
Comment (r14chandra) [reply src/Components/Reports/Common/SystemsFirstPage.test.js:7]: IMO, this code is unclear and complex. Please feel free to update the code as suggested to make it modular.

[the remark this answers is not in the document]
Comment (r14chandra) [reply src/Components/Reports/Common/SystemsFirstPage.test.js:7]: IMO, this code is unclear and complex. Please feel free to update the code as suggested to make it modular.

**Review by r14chandra — CHANGES_REQUESTED** [review]
Overall the PR looks good now. Just suggested one change.

File: src/Components/Reports/Common/SystemsFirstPage.test.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (PreetiW) [reply src/Components/Reports/Common/SystemsFirstPage.test.js:7]: Hello @r14chandra, can you please share what is unclear/complex in the above code? As I would like to keep it as to be able to generate below snapshot. Let me know!

image

[the remark this answers is not in the document]
Comment (r14chandra) [reply src/Components/Reports/Common/SystemsFirstPage.test.js:7]: I missed it!

**Review by r14chandra — APPROVED** [review]
lgtm!

**Pull request merged**